src/services/iasql.ts

```diff
@@ -1151,6 +1152,32 @@ export async function uninstall(moduleList: string[], dbId: string, force = fals
   const remainingModules = existingModules.filter(
     (m: string) => !mods.some(m2 => `${m2.name}@${m2.version}` === m),
   );
+  // See if any modules not being uninstalled depend on any of the modules to be uninstalled
+  const leftoverModules = allInstalledModules.filter(
+    (m: any) => !mods.map(n => `${n.name}@${n.version}`).includes(m.name),
+  );
+  // Because of TypeORM weirdness with self-referential tables, construct the dependencies array
+  // manually. We can do that because we can use the module's dependencies to figure out what they
+  // should be
+  for (const mod of leftoverModules) {
+    const Module: any = Object.values(Modules).find((m: any) => `${m.name}@${m.version}` === mod.name);
+    if (!Module) throw new Error(`Somehow ${mod.name} does not have a corresponding module defined`);
+    mod.dependencies = [];
+    for (const depName of Module.dependencies) {
+      const dep = allInstalledModules.find((m: any) => m.name === depName);
+      if (!dep) throw new Error(`Somehow ${depName} does not have a corresponding module defined`);
+      mod.dependencies.push(dep);
+    }
+  }
+  for (const mod of leftoverModules) {
+    if (
+      mod.dependencies.filter((m: any) => mods.map(n => `${n.name}@${n.version}`).includes(m.name)).length > 0
```

Comment:
I use this `map`ped `mods` in a few places, and in a for loop here, I should gen it once.